scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/TaskKiller.java

```diff
@@ -67,11 +69,26 @@ private void interruptRepeatedly() {
         }
     }
 
+    /**
+     * Interrupts and gives the thread time to cleanup
+     */
+    private void interruptAndWaitCleanupTime() {
+        threadToKill.interrupt();
+        int iterationsToHitWaitingTimeLimit = CLEANUP_SECONDS_AFTER_FIRST_INTERRUPT * 100;
+        for(int i = 0; threadToKill.isAlive() && i < iterationsToHitWaitingTimeLimit ; i++ ) {
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00108)